apps/src/turtle/blocks.js

```diff
@@ -259,35 +259,84 @@ exports.install = function (blockly, blockInstallOptions) {
     }
   };
 
-blockly.Blocks.point_to = {
-    // Block for pointing to a specified direction
-    helpUrl: '',
-    init: function () {
-      this.setHSV(184, 1.00, 0.74);
-      this.appendDummyInput()
-          .appendTitle(msg.pointTo());
-      this.appendDummyInput()
-          .appendTitle(new blockly.FieldTextInput('0', blockly.FieldTextInput.numberValidator), 'DIRECTION')
+  generator.draw_turn_inline = function () {
+    // Generate JavaScript for turning left or right.
+    var value = window.parseFloat(this.getTitleValue('VALUE'));
+    return 'Turtle.' + this.getTitleValue('DIR') +
+        '(' + value + ', \'block_id_' + this.id + '\');\n';
+  };
+
+  function createPointToDirectionBlock(blockName) {
+    return {
+      helpUrl: '',
+      init: function () {
+        this.setHSV(184, 1.00, 0.74);
+        this.appendDummyInput()
+            .appendTitle(msg.pointTo());
+        this.setPreviousStatement(true);
+        this.setInputsInline(true);
+        this.setNextStatement(true);
+        this.setTooltip(msg.pointTo());
+
+        appendToPointToDirectionBlock(blockName, this);
+      }
+    };
+  }
+
+  function appendToPointToDirectionBlock(blockName, block) {
+    if (blockName === 'point_to_direction') {
+      block.appendDummyInput()
+          .appendTitle(new blockly.FieldAngleTextInput('VALUE', '0'),
+              'DIRECTION')
+          .appendTitle(msg.degrees());
+    } else if (blockName === 'point_to_direction_param') {
+      block.appendValueInput('VALUE')
+          .setCheck(blockly.BlockValueType.NUMBER)
+          .addFieldHelper(blockly.BlockFieldHelper.ANGLE_HELPER, {
+            block: block
+          });
```

Comment:
Nit: we can use [ES6 shorthand property names](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Object_initializer#New_notations_in_ECMAScript_2015) for `block`.

```js
.addFieldHelper(blockly.BlockFieldHelper.ANGLE_HELPER, {block});
```

Reply:
This is neat!  Fixed.